lib/utils/filters.js

```diff
@@ -53,39 +57,44 @@ const validateFilters = function(type, filterObj) {
  * @param {Object} filterObj A simple key-value pair object for setting filters.
  * @returns {String} Returns the query header if the filter is valid. Else, it will throw an error.
  */
-const filterBuilder = function(searchType, filterObj=null) {
-    if (filterObj) { validateFilters(searchType, filterObj); }
+const filterBuilder = function (searchType, filterObj = null) {
+	if (filterObj) {
+		validateFilters(searchType, filterObj);
+	}
 
-    let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
+	let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
         Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } `;
 
-    if (searchType === "activity") {
-        baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
+	if (searchType === "activity") {
+		baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
             Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } 
             activities (id: $id`;
-    }
-    if (searchType === "anime") {
-        baseQuery += "media (type: ANIME, search: $search";
-    }
-    if (searchType === "manga") {
-        baseQuery += "media (type: MANGA, search: $search";
-    }
-
-    if (filterObj) {
-        let filterKeys = Object.keys(filterObj);
-    
-        for (let x = 0; x < filterKeys.length; x++) {
-            let filterValue = filterObj[filterKeys[x]];
-
-            if (filterValue instanceof Array) {
-                filterValue = `[${filterValue.join(", ")}]`;
-            }
-
-            baseQuery += `, ${filterKeys[x]}: ${filterValue}`; 
-        }
-    }
-
-    return `${baseQuery}) {`;
+	} else if (searchType === "anime") {
+		baseQuery += "media (type: ANIME, search: $search";
+	} else if (searchType === "manga") {
+		baseQuery += "media (type: MANGA, search: $search";
+	}
```

Comment:
Can you revert this back into the three if statements and not one big if/else if block?

Reply:
hmm, what does this return in benefit, the if else will make it skip the 2nd/3rd if's if the 1st one is true

Reply:
Well, my belief was that the multiple if statements were faster when computed compared to the if/else blocks. I thought that was the case but I can't find anything to back that up that belief anymore... so this ended up as a pointless comment.

This is fine the way it is. 👍